reference/events.rst

```diff
@@ -158,6 +158,35 @@ Listener Class Name
 :class:`Symfony\\Component\\HttpKernel\\EventListener\\StreamedResponseListener`     -1024
 ===================================================================================  ========
 
+``kernel.finish_request``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Event Class**: :class:`Symfony\\Component\\HttpKernel\\Event\\FinishRequestEvent`
+
+The purpose of this event is to allow you to reset the global and environmental state of
+the application after a sub-request has finished::
```

Comment:
I suggest to change this to "after a sub-request has finished. For example, the translator listener uses it to reset the locale to the locale of the parent request:". However, that can also be done by us after merging, so don't worry if you have no time to update the PR.